javascript/javascript2/week2/1.js

```diff
@@ -0,0 +1,34 @@
+// you would like to write a program that doubles the odd numbers in an array and throws away the even number.
+
+let numbers = [1, 2, 3, 4];
+
+let newNumbers = [];
+for (let i = 0; i < numbers.length; i++) {
+  if (numbers[i] % 2 !== 0) {
+    newNumbers[i] = numbers[i] * 2;
+  }
+}
+
+console.log("The doubled numbers are", newNumbers); // [2, 6]
+
+//Rewrite the above program using map and filter don't forget to use arrow functions.
+
+const newNumbersArray = numbers.filter(function(n) {
+  return n % 2 === 0;
+});
+
+console.log(newNumbersArray);
+
+function multByTwo(number) {
+  return number * 2;
+}
+const multByTwoArray = newNumbersArray.map(multByTwo);
+console.log(multByTwoArray);
+
+//// OR
+
+const newNumbersArray2 = numbers.filter(n => n % 2 === 0);
```

Comment:
good. remember you can chain this functions like this. can you explain why the code bellow works?
`result = numbers.filter(number => {return number%2 != 0}).map( number => {return 2*number})
`

Reply:
Hi Florin. 
result = numbers
              .filter(number => {return number%2 != 0})
              .map( number => {return 2*number})
This code do the same, first step it filtrs and takes just numbers which can be divide by 2 and do not equal 0. And then this result ( each number) multiplys by 2

Reply:
Your code is correct,yes. I was pointing out that you can make it shorter, without extra variables

Reply:
yes, yes, it's definitely shorter)) thank you!